opsgenie/resource_opsgenie_role.go

```diff
@@ -0,0 +1,220 @@
+		return output
+	}
+
+	for _, v := range input.List() {
+		output = append(output, v.(string))
+	}
+	return output
+}
+
+func resourceOpsGenieCustomUserRoleCreate(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+
+	if err != nil {
+		return err
+	}
+
+	UserRoleName := d.Get("rolename").(string)
+	ExtendedUserRole := d.Get("extended_role").(string)
+	GrantedRights := flattenSet(d.Get("granted_rights").(*schema.Set))
+	DisallowedRights := flattenSet(d.Get("disallowed_rights").(*schema.Set))
+
+	log.Printf("[INFO] Creating OpsGenie custom user role '%s'", UserRoleName)
+	result, err := client.Create(context.Background(), &custom_user_role.CreateRequest{
+		Name:             UserRoleName,
+		ExtendedRole:     custom_user_role.ExtendedRole(ExtendedUserRole),
+		GrantedRights:    GrantedRights,
+		DisallowedRights: DisallowedRights,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(result.Id)
+
+	return resourceOpsGenieCustomUserRoleRead(d, meta)
+}
+
+func resourceOpsGenieCustomUserRoleRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	UserRoleName := d.Get("rolename").(string)
+
+	log.Printf("[INFO] Reading OpsGenie custom role '%s'", UserRoleName)
+
+	usrRole, err := client.Get(context.Background(), &custom_user_role.GetRequest{
+		Identifier:     UserRoleName,
+		IdentifierType: custom_user_role.Name,
+	})
+	if err != nil {
+		return err
+	}
+
+	d.Set("username", usrRole.Name)
+	d.Set("full_name", usrRole.ExtendedRole)
+	d.Set("role", usrRole.GrantedRights)
+	d.Set("locale", usrRole.DisallowedRights)
+
+	return nil
+}
+
+func resourceOpsGenieCustomUserRoleUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+
+	if err != nil {
+		return err
+	}
+
+	UserRoleName := d.Get("rolename").(string)
+	ExtendedUserRole := d.Get("extended_role").(string)
+	GrantedRights := flattenSet(d.Get("granted_rights").(*schema.Set))
+	DisallowedRights := flattenSet(d.Get("disallowed_rights").(*schema.Set))
+
+	log.Printf("[INFO] Updating OpsGenie custom user role '%s'", UserRoleName)
+
+	_, err = client.Update(context.Background(), &custom_user_role.UpdateRequest{
+		Identifier:       d.Id(),
+		IdentifierType:   custom_user_role.Id,
+		Name:             UserRoleName,
+		ExtendedRole:     custom_user_role.ExtendedRole(ExtendedUserRole),
+		GrantedRights:    GrantedRights,
+		DisallowedRights: DisallowedRights,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceOpsGenieCustomUserRoleDelete(d *schema.ResourceData, meta interface{}) error {
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+
+	log.Printf("[INFO] Deleting OpsGenie custom user role '%s'", d.Get("rolename").(string))
+
+	if err != nil {
+		return err
+	}
```

Comment:
please move err to above log

Reply:
yes, will do

Reply:
Hi, here did you mean that I should log the error using log.Printf(...) before returning it?

Reply:
Never mind, got it